src/libratbag.c

```diff
@@ -1057,6 +1057,14 @@ static inline bool
 resolution_has_dpi(struct ratbag_resolution *resolution,
 		   unsigned int dpi)
 {
+	if (dpi == 0) {
+		if (resolution->is_active) {
+			/* the device would respond with ERR_INVALID_ARGUMENT */
+			log_error(resolution->profile->device->ratbag,"Setting the active resolution to zero dpi is not allowed\n");
+			return false;
+		} else
+			return true;
+	}
```

Comment:
We need a new property in `resolution`, `is_enabled`. Setting the DPI to 0 to disable the resolution is a implementation detail, other devices might do it differently.

Reply:
As far as I understand it, hidpp20 supports either 5 DPIs or just one (not configurable). Since there is no property in the protocol which tells us weather a dpi is enabled or not, I don't think it makes sense to add this. Maybe I'm wrong here, can someone test this with a non-Logitech mouse that uses hidpp20?